File: Pong/ball.py
import pygame as pg
from settings import *
import random

class Ball(pg.sprite.Sprite):
    def __init__(self):
        super().__init__()
        self.ballSize = 20
        self.ballColor = (200, 200, 200)
        self.image = pg.Surface((self.ballSize, self.ballSize))
        #self.image.fill('white')
        pg.draw.ellipse(self.image, self.ballColor, [0, 0, self.ballSize, self.ballSize])
        self.rect = pg.Rect(width / 2, height / 2, self.ballSize, self.ballSize)

        # Ball Movement
        #self.ballSpeed_x = 7 * random.choice((1, -1))
        #self.ballSpeed_y = 7 * random.choice((1, -1))


    #def ballAnimation(self):
        #self.rect.x += self.ballSpeed_x # Connects X of ball and Speed of ball.
        #self.rect.y += self.ballSpeed_y # Connects Y of ball and Speed of ball. 

        # Screen Constraints
        #if self.rect.top <= 0 or self.rect.bottom >= height: # Top of Screen Constraint & Bottom of Screen Constraint
            #self.ballSpeed_y *= -1                           
        #if self.rect.left <= 0 or self.rect.right >= width:  # Left of Screen Constraint & Right of Screen Constraint
            #self.ballSpeed_x *= -1                           


    def update(self):
        #self.ballAnimation()
        pass

File: Browser Games/Pong/ball.js

class Ball {
    constructor({position}) {
        this.position = position;

        // Randomization of ball direction (On Reset)
        const direction = {
            x: Math.random() - 0.5 >= 0 ? -1 : 1, // If -0.5 to 0 returns -1, if 0 to 5 returns 1
            y: Math.random() - 0.5 >= 0 ? -1 : 1, //  If -0.5 to 0 returns -1, if 0 to 5 returns 1
        };

        this.velocity = {x: direction.x, y: direction.y};
        this.width = 10;
        this.height = 10;
    }

    draw() {
        c.fillStyle = 'white';
        c.fillRect(this.position.x, this.position.y, this.width, this.height);
    }

    collisions() {
        // x coordinate screen constraint
        if (this.position.x + this.width + this.velocity.x >= canvas.width || this.position.x + this.width + this.velocity.x <= 0) {
            this.velocity.x = -this.velocity.x
        }

        // y coordinate screen constraint
        if (this.position.y + this.height + this.velocity.y >= canvas.height || this.position.y + this.height + this.velocity.y <= 0) {
            this.velocity.y = -this.velocity.y
        }

    }

    update() {
        this.draw();

        this.collisions();
        

        // Movement
        this.position.x += this.velocity.x; // Aligns x velocity to ball position.
        this.position.y += this.velocity.y; // Aligns y veloctiy to ball position.
    }
}

